components/InterestZoneModal/InterestZoneHeader.tsx

```diff
@@ -34,105 +35,156 @@ export default function ZoneModalHeader({
-            },
-            selected: {
-              backgroundColor: theme.colors.yellow[2],
-              color: theme.colors.yellow[7],
-            },
-          }}
-          onChange={(e) => form.setFieldValue('status', e as Status)}
-          data={[
-            { value: Status.TODO, label: t(`status.${Status.TODO}`) },
-            { value: Status.INPROGRESS, label: t(`status.${Status.INPROGRESS}`) },
-            { value: Status.DONE, label: t(`status.${Status.DONE}`) },
-          ]}
-          defaultValue={t(`status.${zone?.status}`)}></Select>
-      )}
-      <Group spacing='xs'>
-        {modal.type === 'VIEW_ZONE' && (
-          <ActionIcon
-            color='dark'
-            onClick={() => setModal({ ...modal, type: 'EDIT_ZONE' })}
-            size='xl'
+    <>
+      <Group
+        p='md'
+        position='apart'
+        align='center'
+        sx={{ width: '100%', [theme.fn.smallerThan('xs')]: { justifyContent: 'flex-start' } }}>
+        <Group spacing='sm' sx={{ height: '100%' }}>
+          <Button
+            sx={{ paddingLeft: 0, color: 'black' }}
+            color='yellow'
+            size='md'
+            leftIcon={<ArrowLeft size={50} />}
+            onClick={() => {
+              if (modal.type === 'VIEW_ZONE') {
+                setDeletePressed(false);
+                setModal(undefined);
+              }
+              else setConfirmationVisible(true);
+            }}
+            variant='subtle'>
+            <Title
+              sx={{ fontWeight: 600, [theme.fn.smallerThan('xs')]: { display: 'none' } }}
+              order={5}>
+               {t('components.modalManager.interestZoneModal.back')}
+            </Title>
+          </Button>
+        </Group>
+        {modal.type === 'VIEW_ZONE' ? (
+          <Stack spacing={0} align='center'>
+            <Text size='sm'>Status</Text>
+            <Text size='lg' inline sx={{ textTransform: 'uppercase' }}>
+              {t(`status.${zone?.status}`)}
+            </Text>
+          </Stack>
+        ) : (
+          <Select
+            {...form.getInputProps('status')}
             radius='md'
-            variant='outline'>
-            <EditIcon />
-          </ActionIcon>
+            size='md'
+            styles={{
+              input: {
+                backgroundColor: theme.colors.yellow[5],
+                fontSize: theme.fontSizes.md,
+                textTransform: 'uppercase',
+                border: '2px solid black',
+                width: '170px',
+                textAlign: 'center',
+              },
+              dropdown: {
+                border: '2px solid',
+                borderColor: theme.colors.yellow[3],
+              },
+              hovered: {
+                backgroundColor: theme.colors.yellow[1],
+              },
+              selected: {
+                backgroundColor: theme.colors.yellow[2],
+                color: theme.colors.yellow[7],
+              },
+            }}
```

Comment:
The above styling details are complex enough to make the code less readable. Please look into extracting the styling blocks within this file into dedicated constants. I believes the project already includes similar examples.